source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -53,6 +54,16 @@ void Filter::initiateCall(const Http::RequestHeaderMap& headers,
       (*metadata_context.mutable_filter_metadata())[metadata_it->first] = metadata_it->second;
     }
   }
+  // If metadata_context_namespaces is specified, pass matching typed filter metadata to
+  // the ext_authz service as well.
+  const auto& request_typed_metadata =
+      decoder_callbacks_->streamInfo().dynamicMetadata().typed_filter_metadata();
+  for (const auto& context_key : config_->metadataContextNamespaces()) {
+    const auto& metadata_it = request_typed_metadata.find(context_key);
+    if (metadata_it != request_typed_metadata.end()) {
+      (*metadata_context.mutable_typed_filter_metadata())[metadata_it->first] = metadata_it->second;
+    }
+  }
```

Comment:
Can we merge to for-look to only one?

```
for (const auto& context_key : config_->metadataContextNamespaces()) {
    if (const auto& metadata_it = request_metadata.find(context_key); metadata_it != request_metadata.end()) {
      (*metadata_context.mutable_filter_metadata())[context_key] = metadata_it->second;
    }
    
    if (const auto& metadata_it = request_typed_metadata.find(context_key); metadata_it != request_typed_metadata.end()) {
      (*metadata_context.mutable_typed_filter_metadata())[conext_key] = metadata_it->second;
    }
}
```

Reply:
Ack, I was thinking that breaking it into two blocks can improve readability as each block only does one thing but agreed this is cleaner. Thanks!